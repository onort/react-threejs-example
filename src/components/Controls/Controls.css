.container {
  width: 25vw;
  border: 1px solid var(--light-border);
  border-radius: 2px;
  padding: var(--spacer-lg) var(--spacer-xlg);
  height: 100%;
}

.title {
  margin: 0 0 var(--spacer-md);
  padding: 0;
  font-size: var(--text-xlg);
  font-weight: 700;
}

.sizeSlider {
  flex: 3;
  margin-right: var(--spacer-md);
}

.sizeNumberInput {
  flex: 2;
}

@media screen and (max-width: 1380px) and (min-width: 1001px) {
  .container {
    width: 32.5vw;
    padding: var(--spacer-lg) var(--spacer-md);
  }
}

@media screen and (max-width: 1000px) and (min-width: 405px) {
  .container {
    box-sizing: border-box;
    width: 90vw;
    height: 32.5vh;
    margin-top: 2.5vh;
    padding: var(--spacer-lg) var(--spacer-md);
  }
}

@media screen and (max-width: 404px) {
  .container {
    box-sizing: border-box;
    width: 90vw;
    height: 32.5vh;
    margin-top: 2.5vh;
    padding: var(--spacer-md);
  }

  .sizeSlider {
    flex: 2;
  }
}
