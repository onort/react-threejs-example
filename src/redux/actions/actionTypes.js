export const DECREASE_LENGTH = "DECRESE_LENGTH"
export const INCREASE_LENGTH = "INCREASE_LENGTH"
export const SET_LENGTH = "SET_LENGTH"
