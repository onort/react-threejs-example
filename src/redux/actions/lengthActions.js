import { types } from "./"

export const decreaseLength = () => ({ type: types.DECREASE_LENGTH })

export const increaseLength = () => ({ type: types.INCREASE_LENGTH })

export const setLength = length => ({ type: types.SET_LENGTH, length })
