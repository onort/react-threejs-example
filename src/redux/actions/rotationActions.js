import { types } from "./"

export const toggleAutoRotation = () => ({ type: types.TOGGLE_AUTO_ROTATION })
