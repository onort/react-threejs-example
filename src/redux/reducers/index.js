import lengthReducer from "./lengthReducer"
import rotationReducer from "./rotationReducer"
import rootReducer from "./rootReducer"

export default rootReducer
export { lengthReducer, rotationReducer }
