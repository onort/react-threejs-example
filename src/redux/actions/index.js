import * as types from "./actionTypes"
import { decreaseLength, increaseLength, setLength } from "./lengthActions"
import { toggleAutoRotation } from "./rotationActions"

export { decreaseLength, increaseLength, setLength, toggleAutoRotation, types }
