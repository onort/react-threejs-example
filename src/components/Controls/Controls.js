import React, { Component } from "react"
import { connect } from "react-redux"
import PropTypes from "prop-types"

import styles from "./Controls.css"
import { decreaseLength, increaseLength, setLength } from "../../redux/actions"
import { ControlRow, NumberInput, RangeSlider } from "../"

class Controls extends Component {
  handleDecrease = () => this.props.decrease()

  handleIncrease = () => this.props.increase()

  handleInputChange = e => this.props.setLength(parseInt(e.target.value, 10))

  handleSizeDrag = e => this.props.setLength(+e.target.value)

  render() {
    const { length } = this.props
    return (
      <div className={styles.container}>
        <h2 className={styles.title}>Controls</h2>
        <ControlRow label="Side Length">
          <RangeSlider
            className={styles.sizeSlider}
            currentValue={length}
            onDrag={this.handleSizeDrag}
            rangeMax={50}
            rangeMin={0}
          />
          <NumberInput
            className={styles.sizeNumberInput}
            currentValue={length}
            max={50}
            min={1}
            onChange={this.handleInputChange}
            onDecrease={this.handleDecrease}
            onIncrease={this.handleIncrease}
          />
        </ControlRow>
      </div>
    )
  }
}

const mapStateToProps = ({ length }) => ({
  length
})

const mapDispatchToProps = dispatch => ({
  decrease: () => dispatch(decreaseLength()),
  increase: () => dispatch(increaseLength()),
  setLength: length => dispatch(setLength(length))
})

Controls.propTypes = {
  decrease: PropTypes.func.isRequired,
  increase: PropTypes.func.isRequired,
  length: PropTypes.number,
  setLength: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Controls)
