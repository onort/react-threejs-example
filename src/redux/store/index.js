import { compose, createStore } from "redux"

import rootReducer from "../reducers"

const enhancer = compose(
  window &&
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
    window.__REDUX_DEVTOOLS_EXTENSION__()
)

const store = createStore(rootReducer, enhancer)

export default store
