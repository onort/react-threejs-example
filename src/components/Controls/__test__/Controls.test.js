import React from "react"
import { mount } from "enzyme"
import { Provider } from "react-redux"
import configureStore from "redux-mock-store"

import Controls from "../"
import { Toggle, NumberInput } from "../../"
import { types } from "../../../redux/actions"

describe("<Controls />", () => {
  const mockStore = configureStore()
  const initialState = { length: 25, rotation: true }
  let wrapper
  let store

  beforeEach(() => {
    store = mockStore(initialState)
    wrapper = mount(
      <Provider store={store}>
        <Controls />
      </Provider>
    )
  })

  afterEach(() => {
    store = null
    wrapper.unmount()
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should toggle auto-rotation", () => {
    const expectedAction = { type: types.TOGGLE_AUTO_ROTATION }
    wrapper
      .find(Toggle)
      .props()
      .onToggle()
    expect(store.getActions()).toEqual([expectedAction])
  })

  it("should set length", () => {
    const expectedAction = { type: types.SET_LENGTH, length: 20 }
    const changeEvent = { target: { value: 20 } }
    wrapper
      .find(NumberInput)
      .props()
      .onChange(changeEvent)
    expect(store.getActions()).toEqual([expectedAction])
  })

  it("should increment side length", () => {
    const expectedAction = { type: types.INCREASE_LENGTH }
    wrapper
      .find(NumberInput)
      .props()
      .onIncrease()
    expect(store.getActions()).toEqual([expectedAction])
  })

  it("should decrement side length", () => {
    const expectedAction = { type: types.DECREASE_LENGTH }
    wrapper
      .find(NumberInput)
      .props()
      .onDecrease()
    expect(store.getActions()).toEqual([expectedAction])
  })
})
