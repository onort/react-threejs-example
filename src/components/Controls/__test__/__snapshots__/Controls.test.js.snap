// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Controls)>
    <Controls
      decrease={[Function]}
      increase={[Function]}
      length={25}
      rotation={true}
      setLength={[Function]}
      toggleAutoRotation={[Function]}
    >
      <div
        className="container"
      >
        <h2
          className="title"
        >
          Controls
        </h2>
        <ControlRow
          inline={false}
          label="Side Length"
        >
          <div
            className="row "
          >
            <span
              className="label"
            >
              Side Length
            </span>
            <div
              className="container"
            >
              <RangeSlider
                className="sizeSlider"
                currentValue={25}
                onDrag={[Function]}
                rangeMax={50}
                rangeMin={1}
              >
                <div
                  className="container sizeSlider"
                >
                  <input
                    className="slider"
                    max={50}
                    min={1}
                    onChange={[Function]}
                    type="range"
                    value={25}
                  />
                </div>
              </RangeSlider>
              <NumberInput
                className="sizeNumberInput"
                currentValue={25}
                max={50}
                min={1}
                onChange={[Function]}
                onDecrease={[Function]}
                onIncrease={[Function]}
              >
                <div
                  className="container sizeNumberInput"
                >
                  <input
                    className="numberInput"
                    onChange={[Function]}
                    type="number"
                    value={25}
                  />
                  <div
                    className="counter"
                  >
                    <span
                      className="increase"
                      onClick={[Function]}
                    >
                      +
                    </span>
                    <span
                      className="decrease"
                      onClick={[Function]}
                    >
                      −
                    </span>
                  </div>
                  <span
                    className="rangeLimits"
                  >
                    (
                    1
                     - 
                    50
                    )
                  </span>
                </div>
              </NumberInput>
            </div>
          </div>
        </ControlRow>
        <ControlRow
          inline={true}
          label="Auto Rotation"
        >
          <div
            className="row inline"
          >
            <span
              className="label"
            >
              Auto Rotation
            </span>
            <div
              className="container"
            >
              <Toggle
                checked={true}
                onToggle={[Function]}
              >
                <div
                  className="container"
                >
                  <label
                    className="switch"
                  >
                    <input
                      defaultChecked={true}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="slider"
                    />
                  </label>
                </div>
              </Toggle>
            </div>
          </div>
        </ControlRow>
      </div>
    </Controls>
  </Connect(Controls)>
</Provider>
`;

exports[`<Controls /> should match snapshot 2`] = `"<div class=\\"container\\"><h2 class=\\"title\\">Controls</h2><div class=\\"row \\"><span class=\\"label\\">Side Length</span><div class=\\"container\\"><div class=\\"container sizeSlider\\"><input type=\\"range\\" min=\\"1\\" max=\\"50\\" class=\\"slider\\" value=\\"25\\"></div><div class=\\"container sizeNumberInput\\"><input class=\\"numberInput\\" type=\\"number\\" value=\\"25\\"><div class=\\"counter\\"><span class=\\"increase\\">+</span><span class=\\"decrease\\">−</span></div><span class=\\"rangeLimits\\">(1 - 50)</span></div></div></div><div class=\\"row inline\\"><span class=\\"label\\">Auto Rotation</span><div class=\\"container\\"><div class=\\"container\\"><label class=\\"switch\\"><input type=\\"checkbox\\" checked=\\"\\"><span class=\\"slider\\"></span></label></div></div></div></div>"`;
